provider/oauth2.go

```diff
@@ -110,6 +109,11 @@ func (p Oauth2Handler) LoginHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	elems := strings.Split(r.URL.Path, "/")
```

Comment:
I'd suggest moving this to a method, i.e. `(p Oauth2Handler) makeRedirURL(path string)`. This will allow to get the url from other place